### DISCUSSION TITLE
PAG.PAGView.init（）第一次可以加载pag，但是第二次加载就会报‘Make context current fail’的错误，使用pagView.stop()和pagView.destroy()方法都不能解决报错

### DISCUSSION THREAD

**Discussion by jan0192** [Tencent/libpag#1891] [General]
PAG.PAGView.init（）第一次可以加载pag，但是第二次加载就会报‘Make context current fail’的错误，使用pagView.stop()和pagView.destroy()方法都不能解决报错

**Comment by zhalice2011** [comment]
你好, 请问是什么平台, 方便提供复现demo吗

**Reply by mhangs111** [reply]
我也遇到了 小程序版本的

**Comment by jan0192** [comment]
微信小程序
save() { 
				//关于保存，您可以采取您自己的方案，因为gesti只帮您进行了canvas的基本渲染操作。
				let that = this;
				this.cancelSelectedAll();
				setTimeout(() =&gt; uni.canvasToTempFilePath({
					canvasId: "myCanvas",
					canvas: this.canvas,
					fileType: "png",
					quality: 1,
					success: async (res) =&gt; {
						// console.log("成功",res)
						// this.result = res.tempFilePath;
						that.imgList[that.pagindex] = res.tempFilePath
						
						const PAG = await PAGInit({locateFile: (file) =&gt; '/uni_modules/lime-pag/static/' + file})
						const image = await new Promise((resolve) =&gt; {
									uni.getImageInfo({
										src: 'https://pagtest.zhalice.com/assets/cat.png',
										success: async function(ress) {
											const image = await new Promise((resolve) =&gt; {
												const image = that.pagcanvas.createImage();
												image.onload = () =&gt; {
													resolve(image);
												};
												image.src = ress.path;
											});
											resolve(image);	
											
										},
										fail(err) {
											console.log(err);
										}
									});
								});
						let pagimage = await PAG.PAGImage.fromSource(image)
						let file = await that.loadFileByRequest(that.pag)
						const pagload =  await PAG.PAGFile.load(file)
						pagload.replaceImage(that.pagindex,pagimage)
						this.pagview = await PAG.PAGView.init(pagload,that.pagcanvas)
						// console.log(pagload.replaceImage);
						console.log(this.pagview.flush());
						this.pagview.flush()
						// this.pagview.play()
					},
					fail: (err) =&gt; {
						console.log("错误", err)
					}
				}), 500);
			},






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Tencent/libpag"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月28日(星期二) 中午12:32
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Tencent/libpag] PAG.PAGView.init（）第一次可以加载pag，但是第二次加载就会报‘Make context current fail’的错误，使用pagView.stop()和pagView.destroy()方法都不能解决报错 (Discussion #1891)





 
你好, 请问是什么平台, 方便提供复现demo吗

**Comment by CodeJhF** [comment]
感谢 @jan0192 的回复，也抱歉我们没有及时跟进这条。

如果问题仍存在，请在最新版本（4.4 / 4.5）上复现并补充版本号与 demo，我们会优先跟进，本条先关闭。（stale-close）
